aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java

```diff
@@ -0,0 +1,157 @@
+package com.redhat.cloud.notifications;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.cloud.notifications.db.EmailAggregationResources;
+import com.redhat.cloud.notifications.helpers.ResourceHelpers;
+import com.redhat.cloud.notifications.models.AggregationCommand;
+import io.quarkus.test.TestTransaction;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.smallrye.reactive.messaging.connectors.InMemoryConnector;
+import io.smallrye.reactive.messaging.connectors.InMemorySink;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static com.redhat.cloud.notifications.EmailSubscriptionType.DAILY;
+import static java.time.ZoneOffset.UTC;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@QuarkusTestResource(TestLifecycleManager.class)
+class DailyEmailAggregationJobTest {
+
+    @Inject
+    ResourceHelpers helpers;
+
+    @Inject
+    DailyEmailAggregationJob testee;
+
+    @Inject
+    @Any
+    InMemoryConnector connector;
+
+    @BeforeEach
+    void setUp() {
+        helpers.purgeEmailAggregations();
+    }
+
+    @Test
+    @TestTransaction
+    void shouldSentTwoAggregationsToKafkaTopic() {
+        System.setProperty("notifications.aggregator.email.subscription.periodic.cron.enabled", "true");
```

Comment:
There's no guarantee tests will always be run in the same order from one JUnit version to another. It would be safer to use this in **all** tests (not only the first one) that require the cron job enabled:

```
System.setProperty("notifications.aggregator.email.subscription.periodic.cron.enabled", "true");
// test content
System.clearProperty("notifications.aggregator.email.subscription.periodic.cron.enabled");
```

Reply:
i am clearing system properties before and after each test execution now.